src/Reveal.Sdk.Dom.Tests/Visualizations/Extensions/IVisualizationExtensionsFixture.cs

```diff
@@ -0,0 +1,335 @@
+﻿using Moq;
+using Newtonsoft.Json;
+using Reveal.Sdk.Dom.Filters;
+using Reveal.Sdk.Dom.Visualizations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Newtonsoft.Json;
+using System.Reflection;
+
+namespace Reveal.Sdk.Dom.Tests.Visualizations.Extensions
+{
+    public class IVisualizationExtensionsFixture
+    {
+        [Fact]
+        public void SetPosition_UpdateRowAndColumnSpan_WithoutCondition()
+        {
+            // Arrange
+            var mockVisualization = new Mock<IVisualization>();
+            mockVisualization.SetupProperty(v => v.RowSpan);
+            mockVisualization.SetupProperty(v => v.ColumnSpan);
+            var visualization = mockVisualization.Object;
+
+            var rowSpan = 10;
+            var columnSpan = 20;
+
+            // Act
+            var updatedVisualization = visualization.SetPosition(rowSpan, columnSpan);
+
+            // Assert
+            Assert.Equal(rowSpan, visualization.RowSpan);
+            Assert.Equal(columnSpan, visualization.ColumnSpan);
+            Assert.Same(visualization, updatedVisualization);
+        }
+
+        // TODO: We should find a faster way to create Generic constructor / method 
+        [Theory]
+        [InlineData(typeof(TextFilter), typeof(TextField))]
+        [InlineData(typeof(DateTimeFilter), typeof(DateTimeField))]
+        [InlineData(typeof(NumberFilter), typeof(NumberField))]
+        [InlineData(typeof(TimeFilter), typeof(TimeField))]
+        [InlineData(typeof(DateTimeFilter), typeof(DateField))]
+        public void AddDataFilter_UpdateDataDefinition_ForTabularDataDefinition(Type filterType, Type fieldType)
+        {
+            // Arrange
+            var mockVS = new Mock<IVisualization>() { };
+
+            var dataDefinition = new TabularDataDefinition();
+
+            var fieldConstructor = fieldType.GetConstructor(new[] { typeof(string) });
+            var fieldObj = fieldConstructor.Invoke(new object[] { "testField" });
+            var field = (fieldType.IsAssignableFrom(fieldObj.GetType())) ? (dynamic)fieldObj : null;
+
+            dataDefinition.Fields.Add((IField)field);
+
+            mockVS.SetupGet(p => p.DataDefinition).Returns(dataDefinition);
+            var visualization = mockVS.Object;
+
+            var filterConstructor = filterType.GetConstructor(Type.EmptyTypes);
+            var filter = filterConstructor.Invoke(new object[] { });
+
+            var vsDataDefinitionFields = (visualization.DataDefinition as TabularDataDefinition).Fields;
+
+            foreach (var f in vsDataDefinitionFields)
+            {
+                // Act
+                var addDataFilterMethod = typeof(IVisualizationExtensions).GetMethod(nameof(IVisualizationExtensions.AddDataFilter));

```

Comment:
I find it quite hard to fully understand what is this part is about at first glance, which is not good for a test. I think you should create a helper function. Then in here, call it so that it look meaningful.